core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/UnboundRestartPatchService.java

```diff
@@ -0,0 +1,107 @@
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.flow.service.FlowService;
+
+@Service
+public class UnboundRestartPatchService extends ExistingStackPatchService {
+
+    private static final String AFFECTED_STACK_VERSION = "7.2.11";
+
+    private static final Set<String> AFFECTED_IMAGE_IDS = Set.of(
+            "26cd5a65-cd5c-457d-8d48-9caf1a486516",
+            "19cf97b8-56d8-4be9-b317-998eea99d884",
+            "c24acec3-9110-4474-9082-3620deac0910"
+    );
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UnboundRestartPatchService.class);
+
+    @Inject
+    private StackImageService stackImageService;
+
+    @Inject
+    private ClusterOperationService clusterOperationService;
+
+    @Inject
+    private FlowService flowService;
+
+    @Override
+    public StackPatchType getStackFixType() {
+        return UNBOUND_RESTART;
+    }
+
+    @Override
+    public boolean isAffected(Stack stack) {
+        try {
+            boolean affected = false;
+            if (AFFECTED_STACK_VERSION.equals(stack.getStackVersion())) {
+                Image image = stackImageService.getCurrentImage(stack);
+                affected = AFFECTED_IMAGE_IDS.contains(image.getImageId());
+                LOGGER.debug("Stack {} with version {} and image {} is {} by unbound service restart bug",
+                        stack.getResourceCrn(), stack.getStackVersion(), image.getImageId(), affected ? "affected" : "not affected");
+            } else {
+                LOGGER.debug("Stack {} with version {} is not affected by unbound service restart bug", stack.getResourceCrn(), stack.getStackVersion());
+            }
+            return affected;
+        } catch (CloudbreakImageNotFoundException e) {
+            LOGGER.warn("Image not found for stack " + stack.getResourceCrn(), e);
+            throw new IllegalStateException("Image not found for stack " + stack.getResourceCrn(), e);
+        }
+    }
+
+    @Override
+    void doApply(Stack stack) throws ExistingStackPatchApplyException {
+        if (isCmServerReachable(stack)) {
+            FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(() -> clusterOperationService.updateSalt(stack));
+            Boolean success = Polling.waitPeriodly(1, TimeUnit.MINUTES)
+                    .run(() -> pollFlowState(flowIdentifier));
+            if (!success) {
+                throw new ExistingStackPatchApplyException("Failed to update salt for stack " + stack.getResourceCrn());
+            }
+        } else {
+            LOGGER.info("Salt update cannot run, because CM server is unreachable of stack: {}", stack.getResourceCrn());
+            throw new ExistingStackPatchApplyException("Salt update cannot run, because CM server is unreachable of stack: " + stack.getResourceCrn());
+        }
+    }
+
+    private boolean isCmServerReachable(Stack stack) throws ExistingStackPatchApplyException {
```

Comment:
this logic seems to belong to another CM/instancemetadata related service for future reuse

Reply:
Moved to `stack.getClusterManagerServer()`

Reply:
Thanks, the whole `isCmServerReachable` logic should be reusable, but it could be refactored when needed.